File: big.py
#!/usr/local/bin/python

import os, random, sys
import Image
import cmdline

SWITCHES = ""
OPTIONS = ""

width = 20
HTML = sys.stdout

def Write(outstr, endln='\n'):
    HTML.write(outstr + endln)


if __name__ == '__main__':

    HTML = file('big.html', 'w')
    switch, files = cmdline.CommandLine(SWITCHES, OPTIONS)

    sdict = {}
    ilist = []
    for arg in files:
	img = Image.open(arg)
	img = img.convert('RGB')
	if img.size[0] > 33 or img.size[1] > 16:
	    continue

	if not sdict.has_key(img.size):
	    sdict[img.size] = []
	sdict[img.size].append(arg)
	ilist.append(arg)

    '''
    kl = sdict.keys()
    kl.sort()

    sl = []
    for i in range(0,17):
    	sl.append([0]*34)
    for k in kl:
	sl[k[1]][k[0]] = len(sdict[k])

    print "    ",
    for x in range(11, 34):
	    print "%3d" % x,
    print
    for y in range(9, 17):
	print "%2d :" % y,
	for x in range(11, 34):
	    print "%3d" % sl[y][x],
	print
    '''

    c = y = 0
    Write("<html><body><center><table>")
    while ilist:
	Write("<tr>")
	for x in range(0, width):
	    if not ilist:
		break
	    if x == (width / 2 - 2) and y == (width / 2 - 2):
		Write('<td rowspan=5 colspan=4><center><img src="logo.gif" align=middle></center></td>')
	    elif x >= (width / 2 - 2) and x < (width / 2 + 2) and y >= (width / 2 - 2) and y < (width / 2 + 3):
		pass
	    else:
		f = random.choice(ilist)
		Write('<td><center><img src="%s" align=middle alt="%s"></center></td>' % (f, f))
		ilist.remove(f)
		c = c + 1
	y = y + 1
	Write("</tr>")
    Write("</table></center></body></html>")
    print c, "usable flags found."


#logo.gif: GIF image data, version 89a, 98 x 69

File: big.php
<html>
<head>
<style>
body {background-color: #FFFFFF;}
img {display: block; margin-left: auto; margin-right: auto;}
</style>
</head>
<body>
<center><table>
<?php

$width = 20;
$height = 48;
$xm = 2;
$xp = 2 + ($width % 2);
$ym = 2;
$yp = 3;

function get_images() {
    $ignores = ["logo.gif", "ball.gif", "iso.gif", "fotw.gif"];
    $imgs = array();
    if ($dh = opendir(getcwd())) {
        while (($fn = readdir($dh)) !== false) {
	    $sizes = getimagesize($fn);
	    if (substr($fn, -4) == '.gif' && ($sizes[0] <= 32) && ($sizes[1] <= 16) && !in_array($fn, $ignores))
		$imgs[] = $fn;
        }
        closedir($dh);
    }
    return $imgs;
}

$imgs = get_images();
shuffle($imgs);
//sort($imgs);
$y = 0;
while (count($imgs) >= $width) {
    echo "<tr>\n";
    for ($x = 0; $imgs && ($x < $width); $x = $x + 1) {
	if ($x == floor($width / 2 - $xm) && $y >= floor($width / 2 - $ym) && $y < floor($width / 2 + $yp)) {
	    if ($y == floor($width / 2 - $ym))
		echo '<td rowspan="' . ($ym + $yp) . '" colspan="' . ($xm + $xp) . '"><center><img src="logo.gif" align=middle></center></td>' . "\n";
	    $x = $x + $xp + 1;
	}
	else {
	    $f = array_pop($imgs);
	    echo '<td><img src="' . $f . '" alt="' . $f . '"></td>' . "\n";
	}
    }
    $y = $y + 1;
    echo "</tr>\n";
    if ($y >= $height)
	break;
}

?>
</table></center>

</body>
</html>
